build.gradle.kts

```diff
@@ -0,0 +1,201 @@
+                compile("org.apache.commons:commons-math3:3.6.1")
+                implementation("org.apache.commons:commons-pool2:2.6.0")
+                implementation("org.apache.commons:commons-text:1.6")
+                implementation("org.apache.ftpserver:ftplet-api:1.1.1")
+                implementation("org.apache.ftpserver:ftpserver-core:1.1.1")
+                implementation("org.apache.geronimo.specs:geronimo-jms_1.1_spec:1.1.1")
+                implementation("org.apache.httpcomponents:httpasyncclient:4.1.4")
+                implementation("org.apache.httpcomponents:httpclient:4.5.7")
+                implementation("org.apache.httpcomponents:httpcore-nio:4.4.11")
+                implementation("org.apache.httpcomponents:httpcore:4.4.11")
+                implementation("org.apache.httpcomponents:httpmime:4.5.7")
+                implementation("org.apache.logging.log4j:log4j-1.2-api:2.11.1")
+                implementation("org.apache.logging.log4j:log4j-api:2.11.1")
+                implementation("org.apache.logging.log4j:log4j-core:2.11.1")
+                implementation("org.apache.logging.log4j:log4j-slf4j-impl:2.11.1")
+                implementation("org.apache.mina:mina-core:2.0.19")
+                implementation("org.apache.rat:apache-rat:0.13")
+                implementation("org.apache.tika:tika-core:1.20")
+                implementation("org.apache.tika:tika-parsers:1.20")
+                implementation("org.apache.velocity:velocity:1.7")
+                implementation("org.apache.xmlgraphics:xmlgraphics-commons:2.3")
+                implementation("org.bouncycastle:bcmail-jdk15on:1.60")
+                implementation("org.bouncycastle:bcpkix-jdk15on:1.60")
+                implementation("org.bouncycastle:bcprov-jdk15on:1.60")
+                implementation("org.brotli:dec:0.1.2")
+                implementation("org.codehaus.groovy:groovy-all:2.4.16")
+                implementation("org.exparity:hamcrest-date:2.0.4")
+                implementation("org.freemarker:freemarker:2.3.28")
+                implementation("org.hamcrest:hamcrest-core:1.3")
+                implementation("org.hsqldb:hsqldb:2.4.1")
+                implementation("org.jacoco:org.jacoco.ant:0.8.3")
+                implementation("org.jdom:jdom:1.1.3")
+                implementation("org.jodd:jodd-core:5.0.6")
+                implementation("org.jodd:jodd-lagarto:5.0.6")
+                implementation("org.jodd:jodd-log:5.0.6")
+                implementation("org.jodd:jodd-props:5.0.6")
+                implementation("org.jsoup:jsoup:1.11.3")
+                implementation("org.mongodb:mongo-java-driver:2.11.3")
+                implementation("org.mozilla:rhino:1.7.10")
+                implementation("org.objenesis:objenesis:2.6")
+                implementation("org.ow2.asm:asm:7.0")
+                implementation("org.slf4j:jcl-over-slf4j:1.7.25")
+                compile("org.slf4j:slf4j-api:1.7.25")
```

Comment:
I don't know Gradle well enough but is there a reason for some being (the new) implementation while some are still compile?

Reply:
Frankly speaking I do not like this section yet.

The idea here is to declare dependency **versions** so these versions are used for all the submodules.

However it looks like Gradle forces to declare the scope in multiple places (at dependency constraint and at dependency use sites). So, technically speaking, `implementation`, `compile` and things like that should be aligned with dependency blogs of other submodules.


`compile` "leaks through" and `implementation` does not leak.
For instance:  `:src:components` module declares `compile(project(":src:core"))` which means every module that depends on `:src:components` would automatically get `:src:core` as a transitive dependency.